src/lib.rs

```diff
@@ -120,9 +120,13 @@ use regex::Regex;
 /// # Ok(())
 /// # }
 /// ```
-pub fn from_str<'a, T>(input: &'a str, regex: &str) -> std::result::Result<T, Error> where T: Deserialize<'a> {
-    let regex = Regex::new(&regex).map_err(Error::BadRegex)?;
-    from_str_regex(input, regex)
+#[inline]
+pub fn from_str<'input, T>(input: &'input str, regex: &str) -> std::result::Result<T, Error>
+where
+    T: Deserialize<'input>,
+{
+    let rex = Regex::new(regex).map_err(Error::BadRegex)?;
```

Comment:
Name shadowing can be confusing, but i don't think this applies to `regex` here and replacing it with `rex` doesn't improve this.